libs/application/template-api-modules/src/lib/modules/templates/announcement-of-death/announcement-of-death.service.ts

```diff
@@ -0,0 +1,241 @@
+import type { Logger } from '@island.is/logging'
+import { TemplateApiModuleActionProps } from '../../../types'
+import {
+  DataUploadResponse,
+  EstateRegistrant,
+  Person,
+  PersonType,
+  SyslumennService,
+} from '@island.is/clients/syslumenn'
+import { Answers as aodAnswers } from '@island.is/application/templates/announcement-of-death/types'
+import { FasteignirApi } from '@island.is/clients/assets'
+import { NationalRegistry, RoleConfirmationEnum, PickRole } from './types'
+import { SharedTemplateApiService } from '../../shared'
+import { EinstaklingarApi } from '@island.is/clients/national-registry-v2'
+import { baseMapper } from './announcement-of-death-utils'
+
+import { isPerson } from 'kennitala'
+
+const UPDATE_APPLICATION = `
+mutation UpdateApplication($input: UpdateApplicationInput!) {
+  updateApplication(input: $input) {
+    id
+  }
+}
+`
+
+@Injectable()
+export class AnnouncementOfDeathService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private readonly nationalRegistryPersonApi: EinstaklingarApi,
+    private readonly syslumennService: SyslumennService,
+    private readonly sharedTemplateAPIService: SharedTemplateApiService,
+    private readonly fasteignirApi: FasteignirApi,
+  ) {}
+
+  async syslumennOnEntry({ application, auth }: TemplateApiModuleActionProps) {
+    const estates = await this.syslumennService.getEstateRegistrant(
+      application.applicant,
+    )
+
+    // On entry shouldn't throw errors but rather be handled
+    // in the frontend
+    if (!estates?.length || estates.length === 0) {
+      return {
+        success: false,
+        estates: [],
+      }
+    }
+
+    // TODO IMPORTANT: Address edge cases for multiple deceased familiy members in
+    //                 the next iteration before feature flag is lifted
+    if (estates.length === 1) {
+      await this.updateEstateAnswer({ application, auth }, estates[0])
+    }
+
+    const relationOptions = (await this.syslumennService.getEstateRelations())
+      .relations
+
+    return {
+      success: true,
+      estates,
+      relationOptions,
+    }
+  }
+
+  async updateEstateAnswer(
+    { application, auth }: TemplateApiModuleActionProps,
+    estate: EstateRegistrant,
+  ) {
+    // Mark answer state
+    estate.assets = estate.assets.map(baseMapper)
+    estate.vehicles = [
+      ...estate.vehicles,
+      ...estate.ships,
+      ...estate.flyers,
+    ].map(baseMapper)
+    estate.estateMembers = estate.estateMembers.map(baseMapper)
+
+    // TODO: Move this to some other function that happens on a transition from
+    // a selection of a deceased relative.
+    // That is: if multiple deceased relatives exist have some selection.
+    // OR: think about a way to select from the mapper.
+    const updatedAnswers = {
+      ...application.answers,
+      ...estate,
+    }
+
+    const updateApplicationResponse = await this.sharedTemplateAPIService
```

Comment:
This should be avoided you are essentially calling the gql api from application-system-api back to itself. Is there a reason this could not be populated purely in a dataprovider with some parameters?

Reply:
We're populating from data coming from a service. Perhaps I didn't take a close enough look, was there a way to populate via services in the providers?

Reply:
Also we're pre-populating answers where they are available. Is this possible in another way?

Maybe it is relevant that this is happening "onEntry" for the initial state, that is to say, this happens before any user interaction becomes available for the initial appliciation state.

Reply:
That is the ´create´ method within the application controller

Reply:
Yeah it would be best if you could just fetch the data you need in a dataprovider and then access it via external data.

Reply:
Alright. It took a while but I rewrote the whole shebang to pull from external data in their respective places for answers :hot_face: 

No more gql being used within application-system-api back to itself :smile: